File: 00-java_to_csharp/Employee.cs
using System;

namespace vererb1_java
{
    public class Employee : IComparable<Employee>
    {
        // Attribute um Angestelltendaten zu speichern
        private String name;
        public String Name
        {
            set => name = value;
            get => name;
        }

        // Konstruktoren
        public Employee()
        {
            Name = "Hugo";
        }

        public Employee(String n)
        {
            Name = n;
        }

        public override String ToString()
        {
            var name = this.GetType().Name;
            return Name + " is a " + name + " and earns " + this.Gehalt;
        }

        public decimal Gehalt { get { return this.ComputePay(); } }

        public virtual decimal ComputePay()
        {
            return 0;
        }

        public int CompareTo(Employee other)
        {
            return this.Name.CompareTo(other?.Name);
        }
    }
}

File: 00-java_to_csharp/Employee.java
package Angtst;

public class Employee
{
	// Attribute um Angestelltendaten zu speichern
  private String name;


 		// Konstruktoren
  public Employee()
  {
    	name = "Hugo";
         // System.out.println ("bin im Employeekonstruktor");
  }
   public Employee(String n)
  {
  	name = n;
        //System.out.println ("bin im Employeekonstruktor Param " + n);
  }
  

   protected String getname ()
  {
        return name;
  }


     public String toString ()
  {
        return name;
  }

   public float computePay()
  {
    return 0;
  }  
}


